File: utils/tags.ts
export const POSITIVE_TAGS = [
  { id: "stayed-hydrated", label: "stayed hydrated" },
  { id: "ate-on-time", label: "ate on time" },
  { id: "slept-well", label: "slept well" },
  { id: "exercised", label: "exercised" },
  { id: "peaceful", label: "peaceful" },
  { id: "low-stress", label: "low stress" },
  { id: "well-rested", label: "well rested" },
  { id: "good-energy", label: "good energy" },
] as const;

export const CONTEXT_TAGS = [
  { id: "tired", label: "tired" },
  { id: "sick", label: "sick" },
  { id: "sore", label: "sore" },
  { id: "restless", label: "restless" },
  { id: "stressed", label: "stressed" },
  { id: "anxious", label: "anxious" },
  { id: "low-mood", label: "low mood" },
  { id: "overwhelmed", label: "overwhelmed" },
  { id: "social-eating", label: "social eating" },
  { id: "traveling", label: "traveling" },
  { id: "busy-day", label: "busy day" },
  { id: "worked-late", label: "worked late" },
  { id: "ate-out", label: "ate out" },
  { id: "skipped-meal", label: "skipped a meal" },
  { id: "ate-fast", label: "ate fast" },
  { id: "late-night-eating", label: "late night eating" },
] as const;

export const ALL_TAGS = [...POSITIVE_TAGS, ...CONTEXT_TAGS] as const;

export type TagId = (typeof ALL_TAGS)[number]["id"];

export function getTagLabel(id: string): string {
  return ALL_TAGS.find((t) => t.id === id)?.label ?? id;
}

File: src/main/kotlin/com/zacksimpson/nourish/data/Tags.kt
package com.zacksimpson.nourish.data

data class Tag(val id: String, val label: String)

val POSITIVE_TAGS = listOf(
    Tag("stayed-hydrated", "stayed hydrated"),
    Tag("ate-on-time", "ate on time"),
    Tag("slept-well", "slept well"),
    Tag("exercised", "exercised"),
    Tag("peaceful", "peaceful"),
    Tag("low-stress", "low stress"),
    Tag("well-rested", "well rested"),
    Tag("good-energy", "good energy"),
)

val CONTEXT_TAGS = listOf(
    Tag("tired", "tired"),
    Tag("sick", "sick"),
    Tag("sore", "sore"),
    Tag("restless", "restless"),
    Tag("stressed", "stressed"),
    Tag("anxious", "anxious"),
    Tag("low-mood", "low mood"),
    Tag("overwhelmed", "overwhelmed"),
    Tag("social-eating", "social eating"),
    Tag("traveling", "traveling"),
    Tag("busy-day", "busy day"),
    Tag("worked-late", "worked late"),
    Tag("ate-out", "ate out"),
    Tag("skipped-meal", "skipped a meal"),
    Tag("ate-fast", "ate fast"),
    Tag("late-night-eating", "late night eating"),
)

val ALL_TAGS = POSITIVE_TAGS + CONTEXT_TAGS

fun getTagLabel(id: String): String = ALL_TAGS.firstOrNull { it.id == id }?.label ?: id
